002-es6-modules.md

```diff
@@ -13,111 +13,165 @@ ES6 modules. At which time this draft would be moved to `ACCEPTED`.
-synchronous.
+module metadata in a [Registry](https://whatwg.github.io/loader/#registry). All actions regarding the Registry can be done
+synchronously, though JS level API uses Promises.
 
 **NOTE:** It is not Node's intent to implement the asynchronous pipeline in the
-Loader specification. There is discussion about including a synchronous
-pipeline in the specification as an addendum.
+WHATWG Loader specification.
 
-## Additional Structures Required
+## 3. Additional Structures Required
 
-### **DynamicModuleRecord**
+### 3.1. **DynamicModuleRecord**
 
-A Module Record that presents a view of an Object for its `[[Namespace]]`
+A Module Record that represents a view of an Object for its `[[Namespace]]`
 rather than coming from an environment record.
 
-All exported values are declarative. That means that they are known after the
-file is parsed, but before it is evaluated.
+DynamicModuleRecord preserves the feature that exported values are known when it comes time for `HostResolveImportedModule` to return. That means that they are known after the
+file is parsed, but before it is evaluated. This behavior is preserved by node synchronously executing CJS files when they are encountered during `HostResolveImportedModule`.
 
-When creating a `DynamicModuleRecord` the list of exports is frozen upon
-construction. No new exports may be added. No exports may be removed.
+When creating a `DynamicModuleRecord` the [`[[Exports]]`](https://tc39.github.io/ecma262/#table-29) is frozen upon construction. No new exports may be added. No exports may be removed. The values of the exports will continue to be mutable however.
 
-## Algorithm
+### 3.1.1 DynamicModuleCreate(O)
 
-When loading a file:
+The abstract operation DynamicModuleCreate with arguments `namespace` is used to allow creation of new DynamicModuleRecords. It performs the following steps:
 
-1. Determine if file is ES or CommonJS (CJS).
-2. If CJS
-  1. Evaluate immediately
-  2. Produce a DynamicModuleRecord from `module.exports`
-3. If ES
-  1. Parse for `import`/`export`s and keep record, in order to create bindings
-  2. Gather all submodules by performing loading dependencies recursively
-    * See circular dep semantics below
-  3. Connect `import` bindings for all relevant submodules (see
-     [ModuleDeclarationInstantiation])
-  4. Evaluate
+1. Let M be a newly created object.
+2. Set M's essential internal methods to the definitions specified in [15.2.1.15 Abstract Module Records](https://tc39.github.io/ecma262/#sec-abstract-module-records)
+3. Set M's [[Namespace]] internal slot to DelegatedModuleNamespaceObjectCreate(`M`, `O`)
+4. Set M's [[Evaluated]] internal slot to **true**
+5. Return M
 
-This still guarantees:
+### 3.2. **DelegatedModuleNamespaceObject**
+
+A ModuleNamespaceObject that performs delegation to an Object when accessing properties. This is used for delegation behavior from CJS `module.exports` when imported by ES modules.
```

Comment:
backticks and what's with the discrepancy with the name in the title, i.e. missing "Delegated"?